src/core/ddsi/src/ddsi_sertype.c

```diff
@@ -274,6 +274,22 @@ uint32_t get_xcdr_version (uint16_t cdr_identifier)
   }
 }
 
+uint32_t ddsi_sertype_enc_id_enc_format (uint16_t cdr_identifier)
+{
+  switch (cdr_identifier)
+  {
+    case CDR_LE: case CDR_BE:
+    case CDR2_LE: case CDR2_BE:
+      return CDR_ENC_FORMAT_PLAIN;
+    case D_CDR2_LE: case D_CDR2_BE:
+      return CDR_ENC_FORMAT_DELIMITED;
+    case PL_CDR2_LE: case PL_CDR2_BE:
+      return CDR_ENC_FORMAT_PL;
+    default:
+      abort ();
```

Comment:
I am probably being paranoid here, but I just want to confirm that `cdr_identifier` is not taken from external input and cannot ever be anything other than the covered cases.

Reply:
Good point! There was an assert on a valid CDR encapsulation identifier, but this should have proper error handling because the value is from external input (and I think was not verified before?). I've refactored these 2 functions a bit in commit b783323